flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/DatabaseCalciteSchema.java

```diff
@@ -80,150 +69,49 @@ public Table getTable(String tableName) {
+			CatalogBaseTable catalogBaseTable, ObjectIdentifier tableIdentifier) {
+		if (isTemporary || catalogBaseTable instanceof QueryOperationCatalogView) {
+			return FlinkStatistic.UNKNOWN();
 		}
-	}
-
-	private Table convertQueryOperationView(QueryOperationCatalogView table) {
-		return QueryOperationCatalogViewTable.createCalciteTable(table);
-	}
-
-	private Table convertSinkTable(ConnectorCatalogTable<?, ?> table) {
-		Optional<TableSinkTable> tableSinkTable = table.getTableSink()
-			.map(tableSink -> new TableSinkTable<>(
-				tableSink,
-				FlinkStatistic.UNKNOWN()));
-		if (tableSinkTable.isPresent()) {
-			return tableSinkTable.get();
+		if (catalogBaseTable instanceof CatalogTable) {
+			return FlinkStatistic.builder()
+				.tableStats(extractTableStats(catalog, tableIdentifier))
+				.build();
 		} else {
-			throw new TableException("Cannot convert a connector table " +
-				"without either source or sink.");
-		}
-	}
-
-	private Table convertSourceTable(
-			ConnectorCatalogTable<?, ?> table,
-			TableStats tableStats) {
-		TableSource<?> tableSource = table.getTableSource().get();
-		if (!(tableSource instanceof StreamTableSource ||
-				tableSource instanceof LookupableTableSource)) {
-			throw new TableException(
-				"Only StreamTableSource and LookupableTableSource can be used in Blink planner.");
-		}
-		if (!isStreamingMode && tableSource instanceof StreamTableSource &&
-			!((StreamTableSource<?>) tableSource).isBounded()) {
-			throw new TableException("Only bounded StreamTableSource can be used in batch mode.");
+			throw new TableException("Unsupported table type: " + catalogBaseTable);
 		}
-
-		return new TableSourceTable<>(
-			tableSource,
-			isStreamingMode,
-			FlinkStatistic.builder().tableStats(tableStats).build(),
-			table);
 	}
 
-	private TableStats extractTableStats(CatalogTable table, ObjectPath tablePath) {
-		TableStats tableStats = TableStats.UNKNOWN;
+	private static TableStats extractTableStats(Catalog catalog,
+			ObjectIdentifier objectIdentifier) {
+		final ObjectPath tablePath = objectIdentifier.toObjectPath();
 		try {
-			// TODO supports stats for partitionable table
-			if (!table.isPartitioned()) {
-				Catalog catalog = catalogManager.getCatalog(catalogName).get();
-				CatalogTableStatistics tableStatistics = catalog.getTableStatistics(tablePath);
-				CatalogColumnStatistics columnStatistics = catalog.getTableColumnStatistics(tablePath);
-				tableStats = convertToTableStats(tableStatistics, columnStatistics);
-			}
-			return tableStats;
+			CatalogTableStatistics tableStatistics = catalog.getTableStatistics(tablePath);
+			CatalogColumnStatistics columnStatistics = catalog.getTableColumnStatistics(tablePath);
+			return convertToTableStats(tableStatistics, columnStatistics);
 		} catch (TableNotExistException e) {
-			throw new TableException(format(
+			throw new ValidationException(format(
 				"Could not access table partitions for table: [%s, %s, %s]",
```

Comment:
why says `Could not access table partitions`? It should be `could not access table`?

Reply:
Should be can not access table statistic